src/graphics/gui.c

```diff
@@ -12,19 +15,64 @@
 int screen_width = INITIAL_SCREEN_WIDTH;
 int screen_height = INITIAL_SCREEN_HEIGHT;
 char title[] = GAME_TITLE;
-
 int state = STATE_START;
 
-void runGameGui(){
+void runGameGui() {
+
     MenusBundle *menu_bundle = initMenusBundle(&screen_width, &screen_height);
+    GameplayData *gameplay_data = NULL;
+
     InitWindow(screen_width, screen_height, title);
+    Vector2 mouse_position = {0.0f, 0.0f};
     SetTargetFPS(60);
-    Vector2 mousePos = {0.0f, 0.0f};
-    Vector2 *mp = &mousePos;
-    while (!WindowShouldClose() && state != STATE_EXIT){
-        handleStateClicks(menu_bundle, mp, &state);
-        handleGUIDraw(menu_bundle, mp, &state, &screen_width, &screen_height);
+
+    // TODO: Move to a different file.
+    AssetManager *assets = (AssetManager*) malloc(sizeof(AssetManager));
+    assets->tile_1 = LoadTexture("assets/tile-1.png");
+    assets->tile_2 = LoadTexture("assets/tile-2.png");
+    assets->tile_3 = LoadTexture("assets/tile-3.png");
+    assets->tile_4 = LoadTexture("assets/tile-4.png");
+    assets->tile_5 = LoadTexture("assets/tile-5.png");
+    assets->tile_opened = LoadTexture("assets/tile-opened.png");
+    assets->tile_closed = LoadTexture("assets/tile-closed.png");
+    assets->tile_flag = LoadTexture("assets/tile-flag.png");
+    assets->tile_bomb = LoadTexture("assets/tile-bomb.png");
+
+    while (!WindowShouldClose() && state != STATE_EXIT) {
+
+        handleStateClicks(menu_bundle, &mouse_position, &state);
+
+        // Pre-drawing operations based on current state.
+        switch (state) {
+            case STATE_GAMEPLAY:
+                handleGameplayActions(gameplay_data, &mouse_position, screen_width, screen_height);
```

Comment:
Careful, you are passing in ints to a function that specifies doubles. I am not sure whether or not this is particularly bad but just something to be wary of

Reply:
Oh, that's something I missed! I'll update those to be `int`s. It's not an issue, C is just going to automatically case it into a double, but it's really bad practice.